Automattic/wp-calypso
# Improve `Points to WPCOM` cta beahviour

In case a domain domain doesn't point to WordPress.com after a successful transfer, we show a `Point to WordPress.com` cta in the domain list.

This cta is currently broken and just takes the user to the domain setting page, without any explanation.

## Proposed Changes
This PR fixes the CTA beahviour and update the destination link depending on whether the domain is using WordPress.com nameservers:
- WordPress.com nameservers -> links to DNS setting page with further details
- External nameservers -> links to domain settings page, with nameservers card expanded

In the first case, we display additional details if the domain is not using default A/CNAME records, providing instructions to reset them. See images below.

![dns-message](https://github.com/user-attachments/assets/495c97c7-210f-493a-97e8-3c1e554e312c)
![dns-records-support](https://github.com/user-attachments/assets/cceb43c1-c587-4eaf-a837-a45b9f243572)

## Testing Instructions

Note: To test this PR you'll a recently transferred domains. As an alternative, you can simulate this forcing api to set `transfer_status` property to `completed`.

### Test 1: use external nameservers
- Apply this patch locally or use the Calypso image link.
- Visit site domains list page and verify that the domains has the `Point to WordPress.com` cta
- Click on the cta and verify that it takes to the setting page, with the nameserver card expanded.

### Test 2: use WordPress.com nameserver but set non default A/CNAME records
- Apply this patch locally or use the Calypso image link.
- Visit site domains list page and verify that the domains has the `Point to WordPress.com` cta
- Click on the cta and verify that it takes to DNS edit page
- Verify that 2 messages are displayed, as in the screenshot above
- Verify that the `Learn more` links open the inline support page, showing the default dns records section.


## Pre-merge Checklist
